aioxmpp/xso/types.py

```diff
@@ -640,6 +641,41 @@ def coerce(self, v):
         return v
 
 
+class JSON(AbstractCDataType):
+    """
+    Character data type for JSON structured data.
+
+    .. versionadded:: 0.11
+
+    Upon deserialisation, character data is parsed as JSON using :mod:`json`.
+    On serialisation, the value is serialised as JSON. This implies that the
+    data must be JSON serialisable, but there is no check for that in
+    :meth:`coerce`, as this check would be (a) expensive to do for nested data
+    structures and (b) impossible to do for mutable data structures.
+
+    Example:
+
+    .. code-block:: python
+
+        class JSONContainer(aioxmpp.xso.XSO):
+            TAG = ("urn:xmpp:json:0", "json")
+
+            data = aioxmpp.xso.Text(
+                type_=aioxmpp.xso.JSON
```

Comment:
The example still assumes the class can be used as type directly.

Reply:
Thanks, I missed that.

Reply:
(Fix pushed.)